packages/multi-test/src/wasm.rs

```diff
@@ -647,48 +666,82 @@ mod test {
+            let contract_addr = keeper
+                .register_contract(
+                    cache,
+                    code_id,
+                    Addr::unchecked("foobar"),
+                    Addr::unchecked("admin"),
+                    "label".to_owned(),
+                    1000,
+                )
+                .unwrap();
 
-        // we can register a new instance of this code
-        let contract_addr = keeper.register_contract(&mut cache, code_id).unwrap();
+            // now, we call this contract and see the error message from the contract
+            let info = mock_info("foobar", &[]);
+            let err = keeper
+                .call_instantiate(
+                    contract_addr.clone(),
+                    cache,
+                    &mock_router(),
+                    &block,
+                    info,
+                    b"{}".to_vec(),
+                )
+                .unwrap_err();
+            // StdError from contract_error auto-converted to string
+            assert_eq!(err, "Generic error: Init failed");
 
-        // now, we call this contract and see the error message from the contract
-        let info = mock_info("foobar", &[]);
-        let err = keeper
-            .call_instantiate(
-                contract_addr,
-                &mut cache,
-                &mock_router(),
-                &block,
-                info,
-                b"{}".to_vec(),
-            )
-            .unwrap_err();
-        // StdError from contract_error auto-converted to string
-        assert_eq!(err, "Generic error: Init failed");
+            // and the error for calling an unregistered contract
+            let info = mock_info("foobar", &[]);
+            let err = keeper
+                .call_instantiate(
+                    Addr::unchecked("unregistered"),
+                    cache,
+                    &mock_router(),
+                    &block,
+                    info,
+                    b"{}".to_vec(),
+                )
+                .unwrap_err();
+            // Default error message from router when not found
+            assert_eq!(err, "cw_multi_test::wasm::ContractData not found");
 
-        // and the error for calling an unregistered contract
-        let info = mock_info("foobar", &[]);
-        let err = keeper
-            .call_instantiate(
-                Addr::unchecked("unregistered"),
-                &mut cache,
-                &mock_router(),
-                &block,
-                info,
-                b"{}".to_vec(),
-            )
-            .unwrap_err();
-        // Default error message from router when not found
-        assert_eq!(err, "cw_multi_test::wasm::ContractData not found");
+            Ok(contract_addr)
+        })
+        .unwrap();
 
-        // and flush
-        cache.prepare().commit(&mut wasm_storage);
+        // verify contract data are as expected
+        let contract_data = CONTRACTS
```

Comment:
We should not do this in test code. Actually WasmKeeper.load_contract() does this.

We should make it public and expose it in the `Wasm` interface (and exposed in `App` as well, so we can use it in higher-level tests).

It might also be nice to have a query that returns an Iterator of all ContractData (but we can add that later if needed). This is a query we have in wasmd, but no need to add it yet